hasher-matcher-actioner/hmalib/common/models/bank.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import typing as t
+from datetime import datetime
+from dataclasses import asdict, dataclass
+import uuid
+from boto3.dynamodb.conditions import Key
+from mypy_boto3_dynamodb.service_resource import Table
+
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.signal_type.signal_base import SignalType
+
+from hmalib.common.models.models_base import DynamoDBItem
+
+"""
+# DynamoDB Table HMABanks
+
+## Models
+Stores banks and bank-members & bank-member signals.
+
+BankMemberSignals are signals extracted from the content in the bank member
+object. However, <signal_id> is unique for a signal_type, signal_value tuple.
+This is enforced by the 4th type of entry in the default table index.
+
+## Indices
+
+1. Default Table Index
+PK                        SK                             Item
+#bank_object              <bank_id>                      Bank
+<bank_id>                 member#<bank_member_id>        BankMember
+<bank_id>                 signal#<signal_id>             BankMemberSignal
+signal_type#<signal_type> signal_value#<signal_value>    <signal_id>
+
+Bank objects are all stored under the same partition key because all bank
+information is expected to be less than 10GB. Also, when a bank is deleted, it
+is removed from the (2) BankNameIndex. To be able to find it, there must be
+another way. That is provided by the "known" PK of #bank_object
+
+2. BankNameIndex
```

Comment:
nit: make it clear what does and doesn't exist yet/post merge (could be mistaken but I am not seeing all these indexes in tf just yet). also if we are just checking if the name is repeated we could use a conditional write.

Reply:
Blaargh! I need stacked PRs! Github, are you listening!

By the time all my upcoming PRs land, this should be the state of the world. I assure you that's top of mind.